### PR TITLE
Add command to see the unresolved transactions

### PR DESCRIPTION
**Pull request by GuptaManan100**
## Description

This PR adds the syntax and implementation for reading all pending unresolved atomic transactions. The two following types of queries are supported - 
1. `show unresolved transactions for ks` - This command gets all the unresolved transactions in the keyspace ks.
2. `show unresolved transactions` - This command gets all the unresolved transactions in the default keyspace, which can be set by the `use ks` command.

## Related Issue(s)

- #16245 

## Checklist

-   [x] "Backport to:" labels have been added if this change should be back-ported to release branches
-   [x] If this change is to be back-ported to previous releases, a justification is included in the PR description
-   [x] Tests were added or are not required
-   [x] Did the new or modified tests pass consistently locally and on CI?
-   [x] Documentation was added or is not required

### CODE REVIEW

File: go/vt/vtgate/engine/fake_vcursor_test.go [hunk 1]
```diff
@@ -826,6 +830,13 @@ func (f *loggingVCursor) ReadTransaction(_ context.Context, _ string) (*querypb.
...
 }
 
+func (f *loggingVCursor) UnresolvedTransactions(_ context.Context, _ string) ([]*querypb.TransactionMetadata, error) {
+	if f.resultErr != nil {
+		return nil, f.resultErr
+	}
+	return nil, nil
+}
+
 // SQLParser implements VCursor
```
Comment (harshit-gangal) [inline go/vt/vtgate/engine/fake_vcursor_test.go:840]: if there is no engine test, this method implementation is not required.

File: go/vt/vtgate/engine/primitive.go [hunk 2]
```diff
@@ -134,6 +134,8 @@ type (
 
 		// ReadTransaction reads the state of the given transaction from the metadata manager
 		ReadTransaction(ctx context.Context, transactionID string) (*querypb.TransactionMetadata, error)
+
+		UnresolvedTransactions(ctx context.Context, keyspace string) ([]*querypb.TransactionMetadata, error)
```
Comment (harshit-gangal) [inline go/vt/vtgate/engine/primitive.go:138]: nit: add a method comment

File: go/vt/vtgate/engine/transaction_status.go [hunk 3]
```diff
@@ -76,25 +77,37 @@ func (t *TransactionStatus) getFields() []*querypb.Field {
 }
 
 func (t *TransactionStatus) TryExecute(ctx context.Context, vcursor VCursor, bindVars map[string]*querypb.BindVariable, wantfields bool) (*sqltypes.Result, error) {
-	transactionState, err := vcursor.ReadTransaction(ctx, t.TransactionID)
+	var transactionStatuses []*querypb.TransactionMetadata
+	var err error
+	if t.TransactionID != "" {
+		var transactionState *querypb.TransactionMetadata
+		transactionState, err = vcursor.ReadTransaction(ctx, t.TransactionID)
+		transactionStatuses = append(transactionStatuses, transactionState)
```
Comment (harshit-gangal) [inline go/vt/vtgate/engine/transaction_status.go:85]: mostly not an issue, but there can be an error as well, so we will append a nil to the slice.

**Review by harshit-gangal — APPROVED** [review]

**Review by harshit-gangal — COMMENTED** [review]
need to fix the existing engine test

**Review by systay — APPROVED** [review]

File: go/vt/vtgate/engine/transaction_status.go (hunk 3 above)
Comment (GuptaManan100) [reply go/vt/vtgate/engine/transaction_status.go:85]: yeah, but before we use the slice, we check for the error and return.

File: go/vt/vtgate/engine/fake_vcursor_test.go (hunk 1 above)
Comment (GuptaManan100) [reply go/vt/vtgate/engine/fake_vcursor_test.go:840]: Added more engine test that are using this.

**Pull request merged**